### ISSUE TITLE
Accept or optionally tolerate ICC profiles with an all-zero version field

### ISSUE THREAD

**Issue by telecos** [awxkee/moxcms#190]
This was reported in Chromium bug https://issues.chromium.org/issues/537416045. Some AVIF files found in the wild contain an embedded ICC profile in a `colr` `prof` box whose header version field at bytes 8..11 is `00 00 00 00`. This is not a valid ICC version, but Chromium's historical skcms path accepts these profiles because it only rejects ICC major versions greater than 4.

moxcms currently rejects the same profile during header parsing because `ProfileVersion::try_from(0x00000000)` returns `CmsError::InvalidProfile`. When Chromium enables its Rust ICC path, this stricter behavior causes existing AVIF files that previously decoded to fail.

The Chromium bug identifies a 564-byte Dropbox-derived black-scaled sRGB profile with description `sRGB IEC61966-2-1 black scaled`. Its version, CMM type, and creator fields are zeroed. The reporter notes that this profile is emitted by Dropbox image export pipelines and can survive into AVIF derivatives generated by image CDNs such as Cloudinary `f_auto`.

## Current behavior

Given an otherwise parseable ICC profile with the version bytes zeroed:

```rust
let mut icc = std::fs::read("valid-profile.icc").unwrap();
icc[8..12].copy_from_slice(&[0, 0, 0, 0]);

let result = moxcms::ColorProfile::new_from_slice(&icc);
assert!(result.is_err());
```

The profile is rejected as `CmsError::InvalidProfile`.

## Behavior in other parsers

The zero version field is non-conformant, but other widely used ICC parsers do not appear to reject a profile solely because the encoded version is
`0x00000000`:

- skcms reads the header version and rejects only when the major version is greater than 4: https://github.com/google/skia/blob/main/modules/skcms/skcms.cc#L1413-L1446
- lcms2 validates/normalizes the encoded version in `_validatedVersion()` and then rejects only versions greater than `0x05000000` in `_cmsReadHeader()`:
  https://github.com/mm2/Little-CMS/blob/master/src/cmsio0.c#L817-L833 and
  https://github.com/mm2/Little-CMS/blob/master/src/cmsio0.c#L894-L904

Those parsers may still reject a malformed profile for other reasons, but the all-zero version field itself is not a hard rejection point in those code paths.

Possible resolutions:

- Accept exactly `0x00000000` by default and treat it as `ProfileVersion::V2_0`.
- Keep strict validation by default, but expose an opt-in lenient parser option or constructor for consumers such as browsers and image decoders.
- If accepting all zero-version profiles is too broad, allow only a known problematic profile fingerprint.

If the lenient behavior is accepted, i think it should stay narrow as we do not have evidences of other use cases: other `v0.x` values such as `0x00100000` should still be rejected, and unsupported versions such as v5/iccMAX should remain rejected.
